internal/server/boltdbstate/config_source.go

```diff
@@ -304,10 +305,28 @@ func (s *State) configSourceIndexInit(dbTxn *bolt.Tx, memTxn *memdb.Txn) error {
 		if err := proto.Unmarshal(v, &value); err != nil {
 			return err
 		}
-		if err := s.configSourceIndexSet(memTxn, k, &value); err != nil {
-			return err
-		}
 
+		// If there are any records whose ID is the name of a plugin, pre-v0.10.4
+		// behavior, then we delete that record from the database, since it is
+		// about to be saved with a hashed ID. This is not very elegant, but
+		// is the simplest way for users to upgrade since custom config sourcer
+		// plugins aren't yet supported, as of 1/10/2023.
+		key := string(k)
+		re := regexp.MustCompile(`aws-ssm|consul|kubernetes|null|packer|terraform-cloud|vault`)
```

Comment:
Nit: I think it's good for performance to compile your regex once at startup, and then use it throughout (example here: https://github.com/hashicorp/waypoint/blob/e19fa4dbdc6a80007a82311790a2e29d766f84fd/internal/cli/help.go#L167)

I don't think it really matters here, because I don't think this is in a hot loop, so feel free to disregard in this case.

Reply:
@izaaklauer thanks for pointing that out! with `bucket.ForEach` we do loop through each record in the bucket, so I'd say it's a "warm" loop 😄 I'm going to set the regex before we enter the loop, but still in this function, since this function is called once at the start up of the server.